### PR TITLE
JAMES-2495 upgrade Junit

### PR DESCRIPTION
**Pull request by trantienduchn**
I will handle Junit upgrade, but I can't push to other people branches, then create my copy of https://github.com/linagora/james-project/pull/1605

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please

Was :green_apple: 

I need execution time to review AND execution logs to approuve this PR

File: mailbox/api/pom.xml [hunk 1]
```diff
@@ -93,6 +93,10 @@
             <artifactId>assertj-guava</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.hamcrest</groupId>
+            <artifactId>java-hamcrest</artifactId>
```
Comment (chibenwa) [inline mailbox/api/pom.xml]: Shouldn't it be in test scope?

**Comment by chibenwa** [comment]
The rebase is needed.

Greping the logs don't show suspicious stuff.

Execution timing "matches" the ones on previous build.

Except the "java-hamcrest" dependecy scope issue I :+1:  this PR

File: mailbox/api/pom.xml (hunk 1 above)
Comment (trantienduchn) [reply mailbox/api/pom.xml]: I updated https://github.com/linagora/james-project/pull/1631/commits/61eba12f45bcc92d1fa28fd2a53b13b8e0065673

**Comment by chibenwa** [comment]
```
[61eba12f45bcc92d1fa28fd2a53b13b8e0065673] [ERROR]   DomainUserMaildirMailboxManagerTest>MailboxManagerTest.creatingConcurrentlyMailboxesWithSameParentShouldNotFail:985 » Execution
```

Maildir mailbox test broke.

Java hamcrest dependency might be needed.

**Comment by trantienduchn** [comment]
I ran maven on my local, test passed. Maybe concurrency problem. test this please

**Comment by trantienduchn** [comment]
```
[61eba12f45bcc92d1fa28fd2a53b13b8e0065673] [INFO] Results:
[61eba12f45bcc92d1fa28fd2a53b13b8e0065673] [INFO] 
[61eba12f45bcc92d1fa28fd2a53b13b8e0065673] [ERROR] Errors: 
[61eba12f45bcc92d1fa28fd2a53b13b8e0065673] [ERROR]   FullUserMaildirMailboxManagerTest>MailboxManagerTest.creatingConcurrentlyMailboxesWithSameParentShouldNotFail:985 » Execution
[61eba12f45bcc92d1fa28fd2a53b13b8e0065673] [INFO] 
[61eba12f45bcc92d1fa28fd2a53b13b8e0065673] [ERROR] Tests run: 123, Failures: 0, Errors: 1, Skipped: 68
```
test this please

**Pull request closed without merging**